src/EditorFeatures/Core/Diagnostics/DiagnosticsClassificationTaggerProvider.cs

```diff
@@ -62,18 +61,61 @@ protected internal override bool IsEnabled
             => !_editorOptionsService.Factory.GlobalOptions.GetOptionValue(DefaultTextViewHostOptions.IsInContrastModeId);
 
         protected internal override bool IncludeDiagnostic(DiagnosticData data)
-            => data.CustomTags.Contains(WellKnownDiagnosticTags.Unnecessary);
+        {
+            // Fading diagnostic might have an 'Unnecessary' custom tag or
+            // unnecessary additional location indices.
+            if (!data.CustomTags.Contains(WellKnownDiagnosticTags.Unnecessary)
+                && !TryGetUnnecessaryLocationIndices(data, out _))
```

Comment:
Why do we need the `TryGetUnnecessaryLocationIndices` check here?

If we don't have the unnecessary custom tag, shouldn't we always return false here?

Reply:
Unnecessary custom tag is for fading the primary location. Some analyzers report diagnostics without this tag but with the property bag and additional locations convey the locations to fade.

Reply:
I thought all analyzers that fade out code will have unnecessary tag on the descriptor itself.

If that's not the case, how did the original code work? It was just `data.CustomTags.Contains(WellKnownDiagnosticTags.Unnecessary)`

Reply:
I think you're correct, and this was originally another bug.

https://github.com/dotnet/roslyn/blob/ec6da663c592238cca8e145044e7410c4ca9213a/src/Analyzers/VisualBasic/Analyzers/UseInferredMemberName/VisualBasicUseInferredMemberNameDiagnosticAnalyzer.vb#L51-L59

This never faded out code, because the descriptor doesn't have "Unnecessary" in its custom tags.